### PR TITLE
add WARP-style scattering to SNPsVariantRecalibrator in GvsCreateFilterSet

### PR DESCRIPTION
**Pull request by rsasch**
- adds a scatter to the `SNPsVariantRecalibrator` call when the number of samples is over a certain threshold
 - separates out "generate tsv/csv files" and "load tsv/csv files into BigQuery" steps of `UploadFilterSetToBQ` into two different tasks, `CreateFilterSetFiles` (scattered) and `UploadFilterSetFilesToBQ` (not)

Closes https://github.com/broadinstitute/dsp-spec-ops/issues/326

- [ ] To do before merging: remove change to `.dockstore.yml`

### CODE REVIEW

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 1]
```diff
@@ -73,10 +76,19 @@ workflow GvsCreateFilterSet {
...
 
+        Int snps_variant_recalibration_threshold = 20000
+    }
+
+    call GetNumSamples {
+        input:
+            fq_sample_table = fq_sample_table,
+            service_account_json = service_account_json,
+            project_id = query_project,
+            dataset_name = default_dataset
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateFilterSet.wdl:91]: I think this is unused (given the fq sample table name and query project)

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 2]
```diff
@@ -138,53 +150,137 @@ workflow GvsCreateFilterSet {
...
+        }
+    }
 
-        indel_recal_file = IndelsVariantRecalibrator.recalibration,
-        indel_recal_file_index = IndelsVariantRecalibrator.recalibration_index,
-        indel_recal_tranches = IndelsVariantRecalibrator.tranches,
+    if (GetNumSamples.num_samples <= snps_variant_recalibration_threshold) {
+        call SNPsVariantRecalibrator as SNPsVariantRecalibratorClassic {
+            input:
+                sites_only_variant_filtered_vcf = MergeVCFs.output_vcf,
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateFilterSet.wdl:232]: GitHub diff is confusing… but is this call the same as the "legacy" call we were making?  no parameter changes, etc right?

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 3]
```diff
@@ -138,53 +150,137 @@ workflow GvsCreateFilterSet {
...
+        }
+    }
 
-        service_account_json = service_account_json,
-        query_project = query_project
-   }
+    call UploadFilterSetFilesToBQ {
+        input:
+            filter_set_name = filter_set_name,
+            file_creation_done = if defined(CreateFilterSetFilesScattered.done) then defined(CreateFilterSetFilesScattered.done) else defined(CreateFilterSetFilesClassic.done),
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateFilterSet.wdl:276]: Is it possible to add a comment as to what this does?

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 4]
```diff
@@ -138,53 +150,137 @@ workflow GvsCreateFilterSet {
...
+                mode = "SNP",
+                disk_size = small_disk,
+                gatk_override = gatk_override
+        }
 
-        snp_recal_file = SNPsVariantRecalibrator.recalibration,
-        snp_recal_file_index = SNPsVariantRecalibrator.recalibration_index,
-        snp_recal_tranches = SNPsVariantRecalibrator.tranches,
+        scatter (idx in range(length(ExtractFilterTask.output_vcf))) {
+            call CreateFilterSetFiles as CreateFilterSetFilesScattered {
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsCreateFilterSet.wdl:209]: I think this doesn't do what we want… the sites_only variants are going to be loaded multiple times (one for each scatter).  Without going too far down the rabbit hole, maybe the thing to do here is to Merge the VCFs coming out of the scatter and then load them the same way we do in the classic mode?

**Review by kcibul — CHANGES_REQUESTED** [review]
It might be nice to have all the WARP-related tasks in that new imported tasks WDL.  These ones came from WARP originally can we put them there also?

- IndelsVariantRecalibrator
- SNPsVariantRecalibrator

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl [hunk 5]
```diff
@@ -11,6 +13,7 @@ workflow GvsCreateFilterSet {
 
         String data_project
         String default_dataset
+        String tmp_output_directory
```
[the remark this answers is not in the document]
Comment (rsasch) [reply scripts/variantstore/wdl/GvsCreateFilterSet.wdl]: changed to `output_directory` (same as in `GvsImportGenomes`)

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl (hunk 2 above)
Comment (rsasch) [reply scripts/variantstore/wdl/GvsCreateFilterSet.wdl:240]: right!

File: scripts/variantstore/wdl/GvsCreateFilterSet.wdl (hunk 1 above)
Comment (rsasch) [reply scripts/variantstore/wdl/GvsCreateFilterSet.wdl]: you are correct — good catch!

**Pull request merged**